packages/react/menu/src/Menu.tsx

```diff
@@ -652,10 +653,24 @@ const MenuItem = React.forwardRef((props, forwardedRef) => {
       menuItem.addEventListener(ITEM_SELECT, (event) => onSelect?.(event), { once: true });
       menuItem.dispatchEvent(itemSelectEvent);
       if (itemSelectEvent.defaultPrevented) return;
-      context.onRootClose();
+      /**
+       * Browsers wait for our handler logic to execute before executing their own implementation
+       * in case we have prevented the default behaviour. Therefore, the root will unmount before
+       * the browser's `pointerup` logic has executed causing `pointerdown` to be stuck in some
+       * browsers. In FF this results in text selection occuring on the page when moving the mouse
+       * after close.
+       *
+       * We move the close logic till after the browser receives the pointerup to avoid this.
+       */
+      rootCloseTimerRef.current = window.setTimeout(context.onRootClose, 0);
     }
   };
 
+  React.useEffect(() => {
+    const rootCloseTimer = rootCloseTimerRef.current;
+    return () => window.clearTimeout(rootCloseTimer);
+  });
```

Comment:
Just wondering why we need the assignment here? is it wrap style thing? also should we provide an empty deps array?

Reply:
Ah oops, should definitely add the dep array. I split it into a variable because of habit 🙈  will inline that too.